web/util/queries.ts

```diff
@@ -96,18 +96,15 @@ const SPEAKER_WITH_SESSIONS = `
   _type,
   _updatedAt,
   "sessions": *[_type == "session" && references(^._id) && !(_id in path("drafts.**"))] {
-    _id,
-    title,
-    duration,
+    "session": { _id, _type, title, duration, slug },
     "programContainingSession": *[_type == "program" && references(^._id)] | order(_createdAt)[0] {
       "programStart": startDateTime,
       sessions[] {
+        _key,
         _type,
         duration,
-        session-> {
-          _id,
-          duration,
-        }
+        durationOverride,
+        session->{ _id },
```

Comment:
I find the session timing code quite difficult to follow with all the nested session objects (due in no small part to the complexity of the program/session setup in the underlying schema, I'm sure), but I believe the removal of `duration` here (in commit https://github.com/sanity-io/structured-content-2022/commit/2fe04133f72fbf0282422eaf57d21511b6c8e385) has caused the listed session times to break.

It looks like they are now ignoring the durations of all preceding sessions, so that only "padding" entries are taken into account.

E.g. on http://localhost:3000/speakers/carrie-hane:
- "The Future of Digital Experience […]": used to show "1:15–2:05 PM PDT", now shows "8:30–9:20 AM PDT" (8:30 is the start time of the program, but the session has 4 sessions before it, so it should start 4h 45min later)
- "Improv Consulting […]" used to show "3:15–4:00 PM PDT" but now shows "9:05–9:50 PM PDT" (which seems offset by the preceding padding entries, but not by the preceding sessions)

Reply:
https://structured-content-2022-ggd84iq0r.sanity.build/speakers/carrie-hane (before)
vs https://structured-content-2022-lgit97v4t.sanity.build/speakers/carrie-hane (after)

Reply:
Yeah, the session-handling code is a bit out of control at this point. One factor contributing to this is the schema, as you mention, wrt. both forcing clients to calculate start/end times and the session-in-session structure where a program has a "session" object which either is a "padding"-type session, or wraps an actual Session. Additionally, The lack of any kind of ORM/codegen/strongly typed client which we could use rather than defining our own types that end up being "out of sync" with the schema is also making things more difficult.

In 799a9cc, I attempt to simplify things a bit. This simplification could probably also be ported over to the Program page, but I didn't want to increase the scope of this PR.

Reply:
Thanks, that looks better :+1: And the session start time is fixed.